kalite/testing/base_environment.py

```diff
@@ -34,11 +35,54 @@ def after_all(context):
 
 
 def before_feature(context, feature):
-    pass
+    if "uses_content_paths" in context.tags:
+        setup_content_paths(context)
 
 
 def after_feature(context, feature):
-    pass
+    if "uses_content_paths" in context.tags:
+        teardown_content_paths(context)
+
+
+@set_database
+def setup_content_paths(context, db):
+    """
+    Creaters available content items and adds their urls to the context object.
+
+    :param context: A behave context, to which the attributes "available_content_path" and "unavailable_content_path"
+        will be added.
+    :return: None
+    """
+    # These paths are "magic" -- the success or failure of actually visiting the content items in the browser
+    # depends on these specific values.
+    context.unavailable_content_path, context.available_content_path = (
+        "khan/foo/bar/unavail",
+        "khan/math/arithmetic/addition-subtraction/basic_addition/addition_1/",
+    )
+
+    update_item(path="khan/math/arithmetic/addition-subtraction/basic_addition/addition_1/",
+                update={"available": True})
+
+    with Using(db, [Item], with_transaction=False):
+        context._unavailabe_item = Item.create(
+            title="Unavailable item",
+            description="baz",
+            available=False,
+            kind="Video",
+            id="3",
+            slug="unavail",
+            path=context.unavailable_content_path
+        )
+
+
+def teardown_content_paths(context):
+    """
+    The opposite of ``setup_content_urls``. Removes content items created there.
+
+    :param context: A behave context, which keeps a reference to the Items so we can clean them up.
+    :return: None.
+    """
+    context._unavailabe_item.delete_instance()
```

Comment:
@MCGallaspy one thing, this deletion is not wrapped in the Using context manager, which is causing an error during teardown.

Not the cause of the test errors, but would be good to cleanup.